### PR TITLE
Secrets Tool

### PR DESCRIPTION
**Pull request by robgil-dds**
Adds the MVP of the secrets-tool.

Secrets tool is utility that provides a framework for doing common commands. 

# Terraform
The terraform subcommand wraps terraform and injects variables sourced from KeyVault. This allows you to pass secrets in to terraform without storing secrets on your host or checking them in to source.
```
secrets-tool terraform plan
secrets-tool terraform apply
```

# KeyVault secrets
This is the secrets management module. Currently it supports adding a secret and listing a secret. 
```
secrets-tool secrets --keyvault <vault url> create --key <key> --value <value>
secrets-tool secrets --keyvault <vault url> list

### CODE REVIEW

**Comment by robgil-dds** [comment]
@dandds I migrated over to pipenv, but I'm concerned it changed and/or pinned the deps for the rest of the project. I definitely need another pair of eyes on this before merging. I also abstracted and cleaned up the code in the terraform commands (removed all the alpha testing type code).

File: Pipfile [hunk 1]
```diff
@@ -4,35 +4,101 @@ verify_ssl = true
...
+six = "==1.13.0"
+SQLAlchemy-Utils = "==0.36.1"
+tabulate = "==0.8.6"
+typed-ast = "==1.4.0"
+urllib3 = "==1.25.7"
+vine = "==1.3.0"
+wrapt = "==1.11.2"
+WTForms = "==2.2.1"
+zipp = "==1.0.0"
+flask-assets = "==2.0"
```
Comment (dandds) [inline Pipfile]: Managing both sets of dependencies out of one Pipfile isn't ideal. Can you try adding a [basic Pipfile](https://pipenv.readthedocs.io/en/latest/basics/#example-pipfile) to the terraform dir and managing the dependencies there? I'd recommend making the dependency list empty in the file and then doing `pipenv install [package]` for each top-level package you import in your python code. `pipenv` will manage sub-dependencies for you.

If you don't have time to tackle this I can push a commit to do it before the start of the day tomorrow.

File: terraform/secrets-tool/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+pyOpenSSL==19.1.0
+python-dateutil==2.8.1
+PyYAML==5.2
+requests==2.22.0
+requests-oauthlib==1.3.0
+six==1.13.0
+tabulate==0.8.6
+typed-ast==1.4.0
+urllib3==1.25.7
+wrapt==1.11.2
```
[the remark this answers is not in the document]
Comment (robgil-dds) [reply terraform/secrets-tool/requirements.txt]: Switches to pipenv

[the remark this answers is not in the document]
Comment (robgil-dds) [reply terraform/secrets-tool/requirements.txt]: Do you want me to write docs for pipenv?

**Pull request merged**